File: magma-platform/magma-platform-bukkit/src/main/java/de/jalumu/magma/platform/paper/text/BukkitNotification.java
package de.jalumu.magma.platform.paper.text;

import de.jalumu.magma.platform.base.text.notification.NotificationBaseImp;
import net.kyori.adventure.key.Key;
import net.kyori.adventure.sound.Sound;

public class BukkitNotification extends NotificationBaseImp {

    private String templateString;
    private String soundString;


    protected void setNotificationTemplate(String template){
        templateString = template;
    }

    protected void setDefaultSound(String sound){
        this.soundString = sound;
    }

    @Override
    public String getNotificationTemplate() {
        return templateString;
    }

    @Override
    public Sound getDefaultSound() {
        return Sound.sound(Key.key(soundString,':'), Sound.Source.MASTER,1f,1f);
    }

}

File: magma-platform/magma-platform-bukkit/src/main/kotlin/de/jalumu/magma/platform/bukkit/text/notification/BukkitNotification.kt
package de.jalumu.magma.platform.bukkit.text.notification

import de.jalumu.magma.platform.base.text.notification.NotificationBaseImp
import net.kyori.adventure.key.Key
import net.kyori.adventure.sound.Sound

class BukkitNotification : NotificationBaseImp() {
    private var templateString: String? = null
    private var soundString: String? = null
    fun setNotificationTemplate(template: String?) {
        templateString = template
    }

    fun setDefaultSound(sound: String?) {
        soundString = sound
    }

    override fun getNotificationTemplate(): String {
        return templateString!!
    }

    override fun getDefaultSound(): Sound {
        return Sound.sound(Key.key(soundString!!, ':'), Sound.Source.MASTER, 1f, 1f)
    }
}